test/test_eos.py

```diff
@@ -24,47 +24,382 @@
+
+    press0 = 101500.0
+    temp0 = 300.0
+    y0s = np.zeros(shape=(nspecies,))
+    for i in range(nspecies-1):
+        y0s[i] = y0 / (10.0 ** (i + 1))
+    spec_sum = sum([y0s[i] for i in range(nspecies-1)])
+    y0s[nspecies-1] = 1.0 - spec_sum
+
+    for fac in range(1, 11):
+        pressin = fac * press0
+        tempin = fac * temp0
+
+        print(f"Testing (t,P) = ({tempin}, {pressin})")
+        cantera_soln = cantera.Solution(phase_id="gas", source=mech_cti)
+        cantera_soln.TPY = tempin, pressin, y0s
+        cantera_soln.equilibrate("UV")
+        can_t, can_rho, can_y = cantera_soln.TDY
+        can_p = cantera_soln.P
+        can_e = cantera_soln.int_energy_mass
+        can_k = cantera_soln.forward_rate_constants
+        can_c = cantera_soln.concentrations
+
+        # Chemistry functions for testing pyro chem
+        can_r = cantera_soln.net_rates_of_progress
+        can_omega = cantera_soln.net_production_rates
+
+        ones = (1.0 + nodes[0]) - nodes[0]
+        tin = can_t * ones
+        pin = can_p * ones
+        yin = make_obj_array([can_y[i] * ones for i in range(nspecies)])
+
+        prom_rho = prometheus_mechanism.get_density(pin, tin, yin)
+        prom_e = prometheus_mechanism.get_mixture_internal_energy_mass(tin, yin)
+        prom_t = prometheus_mechanism.get_temperature(prom_e, tin, yin, True)
+        prom_p = prometheus_mechanism.get_pressure(prom_rho, tin, yin)
+        prom_c = prometheus_mechanism.get_concentrations(prom_rho, yin)
+        prom_k = prometheus_mechanism.get_fwd_rate_coefficients(prom_t, prom_c)
+
+        # Pyro chemistry functions (uncomment to break test)
+        prom_r = prometheus_mechanism.get_net_rates_of_progress(prom_t,
+                                                                prom_c)
+        prom_omega = prometheus_mechanism.get_net_production_rates(prom_rho,
+                                                                   prom_t, yin)
+
+        print(f"can(rho, y, p, t, e, k) = ({can_rho}, {can_y}, "
+              f"{can_p}, {can_t}, {can_e}, {can_k})")
+        print(f"prom(rho, y, p, t, e, k) = ({prom_rho}, {y0s}, "
+              f"{prom_p}, {prom_t}, {prom_e}, {prom_k})")
+
+        # For pyro chem testing
+        print(f"can_r = {can_r}")
+        print(f"prom_r = {prom_r}")
+        print(f"can_omega = {can_omega}")
+        print(f"prom_omega = {prom_omega}")
+
+        assert discr.norm((prom_c - can_c) / can_c, np.inf) < 1e-14
+        assert discr.norm((prom_t - can_t) / can_t, np.inf) < 1e-14
+        assert discr.norm((prom_rho - can_rho) / can_rho, np.inf) < 1e-14
+        assert discr.norm((prom_p - can_p) / can_p, np.inf) < 1e-14
+        assert discr.norm((prom_e - can_e) / can_e, np.inf) < 1e-6
+        assert discr.norm((prom_k - can_k) / can_k, np.inf) < 1e-10
+
+        # Pyro chem test comparisons
+        rate_tol = 1e-12
+        if mechname == "sanDiego":
+            rate_tol = 1e-8
```

Comment:
Maybe parametrize this together with `mechname` as a tuple?

Reply:
I think (722f51d)